docker/development/Dockerfile

```diff
@@ -12,13 +12,37 @@ ENV RAILS_ENV=production
 EXPOSE 3000
 EXPOSE 9394
 
-# https://github.com/nodesource/distributions#installation-instructions
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
-    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
-    curl -sS https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - && echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | tee  /etc/apt/sources.list.d/pgdg.list && \
+# use "--login" option, so that .bashrc is sourced in new shells
+SHELL ["/bin/bash", "--login", "-c"] 
+
+# Install Node.js
+# https://github.com/nodesource/distributions/issues/1583#issuecomment-1597489401
+# https://stackoverflow.com/a/57546198/
+# Unfortunately, we have to explicitly specify the node version here
+# and cannot use 16.x as we need to put the node binary into the PATH
+# and therefore require the exact version to find the folder
+ENV NODE_VERSION=16.20.1
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+ENV NVM_DIR=/root/.nvm
+RUN source ${NVM_DIR}/nvm.sh && nvm install ${NODE_VERSION} && \
+    nvm use ${NODE_VERSION} && nvm alias default ${NODE_VERSION}
+
+ENV NODE_PATH="${NVM_DIR}/versions/node/v${NODE_VERSION}/bin/"
+ENV PATH="${NODE_PATH}:${PATH}"
+
+RUN nvm current
+RUN node --version
+RUN npm --version
+
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && \
+    apt-get install -y lsb-release && \
+    # https://wiki.postgresql.org/wiki/Apt
+    echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" | tee  /etc/apt/sources.list.d/pgdg.list && \
```

Comment:
Why is there an extra repo added for this? Debian bullseye (the current oldstable release) ships version 13, see https://packages.debian.org/search?keywords=postgresql

Reply:
Also the ruby image exists both with `bullseye` and `bookworm`: https://hub.docker.com/_/ruby
- [`3.1.4-bullseye`, `3.1-bullseye`](https://github.com/docker-library/ruby/blob/ed1be47a38a7a24a0aa03c450549afcb592f02a8/3.1/bullseye/Dockerfile)
- [`3.1.4-bookworm`, `3.1-bookworm`, `3.1.4`, `3.1`](https://github.com/docker-library/ruby/blob/ed1be47a38a7a24a0aa03c450549afcb592f02a8/3.1/bookworm/Dockerfile)

So it should be enough to just switch this to the `3.1.4-bookworm` base image or to move everything else forward

Reply:
To be honest, I don't know why we have this statement in here, especially since the docker compose file uses the `postgres:13` image for the `db` service and in production we don't have this service anyways as the database is managed separately. Will try to get rid of it entirely.

Reply:
Ok, I found the reason why we add this repo: we need the package `postgresql-client-13`, which is provided by the APT package we added in the mentioned line. See the [PostgrSQL wiki](https://wiki.postgresql.org/wiki/Apt).

I'm not sure I understand what you mean with the ruby image. Should I specify it instead of `postgres:13` in the `db` service in docker compose? However, we just need the Postgres database here, so I think just installing it alone via the script would be better, wouldn't it? Also, with the script, we can use `$(lsb_release)` to install the package dependent on the distribution as developers might have different distributions.

Reply:
Yes, I also thought that this is the reason why you add the repo. However my point is that this package should be availiable from the official debian repository of the bookworm distribution. There is a version of our base image that is based on bookworm, so we can just use it.

Reply:
Thanks for the tip, works like a charm with `3.1.4-bullseye`, so we don't have to add the PostgreSQL repo manually anymore.